crates/ruff/src/rules/flake8_comprehensions/fixes.rs

```diff
@@ -552,7 +573,51 @@ pub(crate) fn fix_unnecessary_collection_call(
     };
     tree.codegen(&mut state);
 
-    Ok(Edit::range_replacement(state.to_string(), expr.range()))
+    let code = if wrap_in_spaces {
+        wrap_code_in_spaces(&state, checker, expr.range())
+    } else {
+        state.to_string()
+    };
+
+    Ok(Edit::range_replacement(code, expr.range()))
+}
+
+/// Adds spaces around the code generated from `state` if `wrap_in_spaces` is true.
```

Comment:
Nit: I'd probably rather parenthesize than add whitespace.

Reply:
probably it's more robust, but I think it renders uglier expressions in many simple cases, what do you think?